### DISCUSSION TITLE
EM35x devices is removed from EZSP 6.9.0.0 as of 2020-12-09

### DISCUSSION THREAD

**Discussion by MattWestb** [zigpy/zigpy#578] [General]
[Zigbee EmberZNet SDK 6.9.0.0](https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.9.0.0.pdf) 6 Removed Items

Removed in release 6.9.0.0
All support for EM35x devices is removed. Additionally, Lightweight IP and Low-Voltage Shutdown plugins are removed.

And:

5 Deprecated Items
Advanced Notice
Zigbee support – The upcoming Zigbee R23 specification will not be supported on any devices matching the following criteria:
• All EM35x devices
• Any devices with 256 kB flash or less [All wireless MCU families]

So the "Golden Version" for EM35x is being 6.8.2.X then the bug and security fixes is integrated.

I have the paper briefly red and its in many parts referring to `v6.8.n` so very likely not 110% updated / misspelled as v6.8.0.1 was.

The good thing for My IKEA Billys (first gen with 256 kb flash) is that is still supported if I have reading right until R23 is integrated in EZSP so little more bug fixes is coming (and found by Silabs).

And little more Cluster Library futures added "Updated ZCL framework to ZCL 8".
And Green Power Bidirectional Messaging support is added.

**Comment by Hedda** [comment]
> So the "Golden Version" for EM35x is being 6.8.0.X then the bug and security fixes is integrated.

* That should probably 6.8.x.x since the latest 6.8 bug-fix release is EmberZNet 6.8.2.0

https://www.silabs.com/documents/public/release-notes/emberznet-release-notes-6.8.2.0.pdf

**Reply by MattWestb** [reply]
It can coming one later fixed release but not so likely.
Then R23 is released and most first gen EFR32 is ended its more likely releasing one security fix then its much more devices with this chips out in aktive systems but its company policy / customer demands that deciding it.
Edit: Was updating the 6.8.0.X  to 6.8.2.X then it was one typo.

**Reply by MattWestb** [reply]
Sorry i was forgetting saying welcome to the zigpy discussions @Hedda !!

**Comment by MattWestb** [comment]
Its looks that the EZSP 6.7.X is going to being the "golden release" for both EM3XX and first gen EFR32 devices.

Silab have removed the USB-serial support in app builder in 6.7.7.0 that making it more tricky but one of our devs have cocking one 6.7.8.0 firmware for EM35X 6.7.8.0 with USB-serial support that still waiting to being verified to working OK.

EZSP 6.8X and 6.9.X has not getting some critical zigbee bug fixes that 6.7.8.0 have getting so they is **not recommended** for using in ZHA for the moment.

Edit:  One EM357 firmware is made but not verified to working with ZHA.
Hi is doing one try making one EM3581 version but its little more tricky and hi dont have any reference hardware for testing the build on.
